### DISCUSSION TITLE
Best to way to parallelize star alignment

### DISCUSSION THREAD

**Discussion by GACGAMA** [alexdobin/STAR#1193] [General]
Hello everyone!

I'm a newbie at linux, so I tought I should ask the experts around here:

Let's say I have 16 threads and 256gb RAM, which is the best way to parallelize?
I know star can use all threads, but is it more optimal to run, for example, 4 threads in 4 concurrent alignments so more memory is used instead of more cores per sample?

What would be the easier way to do so? Maybe parallel command with 4 jobs and 4 threads | star using 4 threads each?

**Comment by alexdobin** [comment]
Hello @GACGAMA,

This is an interesting question.

You are talking about running multiple samples in parallel, right?
Splitting one sample into several chunks is not going to be worth it, because of the overhead required for splitting/concatenation of the results.
So the question is which is faster: mapping 4 samples in parallel with 4 threads each, or mapping them consecutively with 16 threads.

Theoretically, running with fewer threads per genome copy in RAM should be faster. However, in practice, the difference probably won't be large. It will depend on many particulars of the system, cache, RAM speed, disk speed, etc - so I would recommend benchmarking it on your machine.

Cheers
Alex
